packages/dev/core/src/Meshes/Builders/graesedLineBuilder.ts

```diff
@@ -0,0 +1,526 @@
+     */
+    lazy?: boolean;
+}
+
+/**
+ * Builder class for create GreasedLineMeshes
+ */
+export class GreasedLineMeshBuilder {
+    /**
+     * Creates a GreasedLine mesh
+     * @param name name of the mesh
+     * @param options options for the mesh
+     * @param scene scene where the mesh will be created
+     * @returns instance of GreasedLineMesh
+     */
+    public static CreateGreasedLine(name: string, options: GreasedLineBuilderOptions, scene: Nullable<Scene>) {
+        scene = <Scene>(scene ?? EngineStore.LastCreatedScene);
+
+        let instance;
+        const allPoints = GreasedLineMeshBuilder.ConvertPoints(options.points);
+
+        let length = 0;
+        if (Array.isArray(allPoints[0])) {
+            allPoints.forEach((points) => {
+                length += points.length / 3;
+            });
+        }
+
+        const widths = GreasedLineMeshBuilder.NormalizeWidthTable(length, options.widths ?? [], GreasedLineMeshWidthDistribution.WIDTH_DISTRIBUTION_START);
+
+        const colors = options.colors
+            ? GreasedLineMeshBuilder.NormalizeColorTable(length, options.colors, GreasedLineMeshColorDistribution.COLOR_DISTRIBUTION_START, options.color)
+            : undefined;
+
+        const offsets = options.offsets instanceof Array<Vector3> ? (<Vector3[]>options.offsets).flatMap((v) => [v.x, v.y, v.z]) : options.offsets;
+
+        // create new mesh if instance is not defined
+        if (!options.instance) {
+            const initialGreasedLineOptions: GreasedLineMeshOptions = {
+                points: allPoints,
+                offsets,
+                updatable: options.updatable,
+                widths,
+                widthsDistribution: options.widthsDistribution,
+            };
+
+            const initialMaterialOptions: GreasedLineMaterialOptions = {
+                materialType: options.materialType,
+                colorDistribution: options.colorDistribution,
+                dashArray: options.dashArray,
+                dashOffset: options.dashOffset,
+                dashRatio: options.dashRatio,
+                resolution: options.resolution,
+                sizeAttenuation: options.sizeAttenuation,
+                useColors: options.useColors,
+                useDash: options.useDash,
+                visibility: options.visibility,
+                width: options.width,
+                color: options.color,
+                colorMode: options.colorMode ?? GreasedLineMeshColorMode.COLOR_MODE_SET,
+            };
+
+            if (colors) {
+                initialMaterialOptions.colors = GreasedLineMeshBuilder.Color3toUint8(colors);
+            }
+
+            const material = options.materialType === GreasedLineMeshMaterialType.MATERIAL_TYPE_PBR ? new PBRMaterial(name, scene) : new StandardMaterial(name, scene);
+            const plugin = new GreasedLinePluginMaterial(material, scene, initialMaterialOptions);

```

Comment:
Yes, the material is so specific it can be used only with GreasedLinesMesh.

Reply:
Yes, I understand that, but my question was about reusing the same greased line material for multiple greased line meshes: isn't that possible? It would seem that if multiple greased line meshes use the same parameters for their material, the same material could be reused for all of them?

Reply:
Another solution has just popped out of my mind. What if I create a public static function in `GreasedLineBuilder` called `CreateGreasedLineMaterial` and add the `material` option to the `CreateGreasedLineMesh` function as mandatory.  So at first the user creates the material. In the second step the user creates the mesh with that material, or more meshes with the same material.